feat(addScript): disable `'use strict'` to allow setting globals

timpur opened the pull request.

timpur commented:
@d3viant0ne @bebraw @SpaceK33z @TheLarkInn

antonyoneill commented:
Hey @bebraw, @TheLarkInn @d3viant0ne, @evilebottnawi 

What is it going to take to get some love for this repository?

This PR should be considered for merge, as should the documentation one #45.

michael-ciniawsky commented:
This may need to be an option, e.g `options.strict = true || false`, instead if being the default. Do we have an overview of potential security implications?

michael-ciniawsky commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "script-loader",
-  "version": "0.7.2",
+  "version": "0.7.3",
```
Please revert this as it is the maintainers responsibility to do this

timpur replied:
sorry, my bad, will do

michael-ciniawsky commented:
@timpur What's the status here? @timpur Will you continue this PR (no pressure, but please indicated if you're still interested in this)?

timpur commented:
@michael-ciniawsky, i am just not sure how to do the option thing that you requested tbh...

antonyoneill commented:
Hey @timpur take a look at how @jiaweihli has done similar in this PR https://github.com/webpack-contrib/script-loader/pull/31/files#diff-168726dbe96b3ce427e7fedce31bb0bc

michael-ciniawsky commented:
Yep, e.g https://github.com/webpack-contrib/script-loader/pull/31/files#diff-168726dbe96b3ce427e7fedce31bb0bc is a good starting point, also feel free to ask questions and push the code if you're stuck somewhere :)

antonyoneill commented:
Hey guys, let me know if there's anything I can do to get this shepherded into the project.

timpur commented:
Some one want to take over this PR ?

FilippoPolo commented:
Hi all,
I've sent @timpur a PR that adds an option to his logic, so that existing behavior is preserved for current users. You would add ?stripUseStrict to trigger the feature.

timpur commented:
update the pr and added documentation, changed the option to` useStrict: false` to disabled use strict, enabled by default. Can change interface if you dont like.

gfortaine commented:
Hello @michael-ciniawsky

Any plan to merge this PR any time soon, please ? Indeed, as others mentioned, it has proven very useful in some cases https://github.com/webpack-contrib/script-loader/pull/51

evilebottnawi commented:
/cc @mistic can we merge this?

mistic commented:
@evilebottnawi merged
